avr-specs/sync-from-upstream.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+
+import subprocess
+import json
+import copy
+import os
+
+SPECS = {
+    "atmega32u4": {
+        "cpu": "atmega32u4",
+        "no-default-libraries": False,
```

Comment:
looks like is required to pass

Reply:
Suggestion: Add a second global for the common changes:

```python
COMMON = {
    "no-default-libraries": False,
}

# ...

spec.update(COMMON)
spec.update(settings)
```

Reply:
BTW, the reason we need `"no-default-libraries": False,` is that we're currently relying on the avr-gcc "runtime" instead of doing it all in rust.  The long-term goal is of course to move away from that by providing our own custom runtime.